### DISCUSSION TITLE
MongoDB - Date and Number fields are showing up as text

### DISCUSSION THREAD

**Discussion by mehrashiv** [Budibase/budibase#6874] [Help] [answered]
Dear community,
I am trying to build an expense app with mongodb. I have an issue with field type. Following is what I am trying to do

1. Using Mongo Atlas - Created a document with 5 fields (**date** of type date, **category** and **subcategory** of type string, **expense** of type Double and **description** of type string. Please see screenshot#1
2. I was able to create a query in BB to read all the documents as seen in screenshot#2
3. The schema shows the right value except for date. date shows up as text and I changed it to Datetime and saved the query. I can load this data in a table  
4. I created another query to insterOne document. I created bindings and ran the query. the response is result by the schema is now gone from the earlier read query. screenshot#4
5. I created a form and added the date, text and number fields. I used the create query in the form. But when I choose the date or expense field I don't see the option to choose anything. If I choose the text field, I can see all the 5 fields. So it seems the date and expense are seen as text. please see screenshot#5 and#6
6. I did go adn run the get query again and edited the schema for date and number. But it did not help. I still cannot add date and expense field. Would appreciate your help.

![image](https://user-images.githubusercontent.com/9964632/180697333-9f349b99-a784-4770-9055-8c01471fab26.png)
![image](https://user-images.githubusercontent.com/9964632/180695164-300f4a9e-b112-42e7-9528-10de50b97c07.png)
![image](https://user-images.githubusercontent.com/9964632/180695290-c879114b-6472-4870-88fd-0cf8de24d289.png)
![image](https://user-images.githubusercontent.com/9964632/180696187-3b14d6ea-bda5-472d-b73b-4a2471b9ee79.png)
![image](https://user-images.githubusercontent.com/9964632/180696591-64eedb30-7a1c-44ab-be32-9a04cb918869.png)
![image](https://user-images.githubusercontent.com/9964632/180696694-5f5dd75c-6c3a-4519-83f9-04f17a5dd7e5.png)

**Comment by melohagan** [comment]
Hi @mehrashiv 

If you select the *Datetime* option from the dropdown in the schema tab and hit **Save** then it should show up as an option for the *Date Picker* component.

**Reply by mehrashiv** [reply]
Hi @melohagan ,
Thank you for the response. I have done that. But I still cannot get the date and number fields when i create those components.

I am running selfhosted app version 1.1.17.

i can give it a shot with the cloud version and revert.

Thank you,
Shiv

**Reply by mehrashiv** [reply]
Hi @melohagan,
I am running in to the same issue with the cloud version. The date and number fields are showing up as text so I cannot add those components.
![image](https://user-images.githubusercontent.com/9964632/180822563-9843ea00-e904-4c19-89c7-52e105e7c96c.png)
![image](https://user-images.githubusercontent.com/9964632/180822642-042e1e76-430c-4cf2-a6a4-c799eea27ea3.png)

![image](https://user-images.githubusercontent.com/9964632/180822689-3691f0b5-724e-4144-bf70-de06d009ff32.png)

**Comment by mehrashiv** [comment, marked as answer]
Hi @melohagan,
The cloud version did not work. Should I convert this to an issue?

**Reply by melohagan** [reply]
Hey @mehrashiv No it's okay - I've added it to an existing issue here: https://github.com/Budibase/budibase/issues/6778#issuecomment-1194284701

I'll take a look at it today and hopefully get it sorted.

**Reply by melohagan** [reply]
@mehrashiv Just realised this is by design. It was raised before: https://github.com/Budibase/budibase/issues/6238

I need to update the docs because it is a bit confusing. Basically when using custom/rest queries the schema doesn't apply to forms. You can't use the Save Row action with custom queries, so the fields aren't useful in the context of a form.

**Reply by mehrashiv** [reply]
Thanks @melohagan .
So to create a new row with custom query that has a date field, how do I go about choosing a date component? Or is there some other way to do.

**Reply by melohagan** [reply]
Hey @mehrashiv You would just create a form with a Date Picker component, and give the date picker field a unique name.

**Reply by mehrashiv** [reply]
That worked..thank you so much @melohagan. Appreciate your promptness and help.
